src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -125,61 +129,49 @@ public async Task<Checksum> GetChecksumAsync(ProjectId projectId, CancellationTo
                 var orderedProjectIds = GetOrCreateSortedProjectIds(this.ProjectIds);
-                var projectChecksumTasks = orderedProjectIds
-                    .Select(id => (state: this.ProjectStates[id], mustCompute: projectsToInclude == null || projectsToInclude.Contains(id)))
-                    .Where(t => RemoteSupportedLanguages.IsSupported(t.state.Language))
-                    .Select(async t =>
-                    {
-                        // if it's a project that's specifically in the sync'ed cone, include this checksum so that
-                        // this project definitely syncs over.
-                        if (t.mustCompute)
-                            return await t.state.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
-
-                        // If it's a project that is not in the cone, still try to get the latest checksum for it if
-                        // we have it.  That way we don't send over a checksum *without* that project, causing the
-                        // OOP side to throw that project away (along with all the compilation info stored with it).
-                        if (t.state.TryGetStateChecksums(out var stateChecksums))
-                            return stateChecksums;
-
-                        // We have never computed the checksum for this project.  Don't send anything for it.
-                        return null;
-                    })
-                    .ToArray();
-
-                var serializer = this.Services.GetRequiredService<ISerializerService>();
-                var attributesChecksum = this.SolutionAttributes.Checksum;
 
-                var analyzerReferenceChecksums = ChecksumCache.GetOrCreateChecksumCollection(
-                    this.AnalyzerReferences, serializer, cancellationToken);
+                using var _ = ArrayBuilder<Task<ProjectStateChecksums>>.GetInstance(out var projectChecksumTasks);
 
-                var allResults = await Task.WhenAll(projectChecksumTasks).ConfigureAwait(false);
-                using var _1 = ArrayBuilder<ProjectId>.GetInstance(allResults.Length, out var projectIds);
-                using var _2 = ArrayBuilder<Checksum>.GetInstance(allResults.Length, out var projectChecksums);
-                foreach (var projectStateChecksums in allResults)
+                foreach (var orderedProjectId in orderedProjectIds)
                 {
-                    if (projectStateChecksums != null)
-                    {
-                        projectIds.Add(projectStateChecksums.ProjectId);
-                        projectChecksums.Add(projectStateChecksums.Checksum);
-                    }
+                    var projectState = this.ProjectStates[orderedProjectId];
+                    if (!RemoteSupportedLanguages.IsSupported(projectState.Language))
+                        continue;
+
+                    if (!projectCone.Object.Contains(orderedProjectId))
+                        continue;
```

Comment:
liek before, we only include projects in the cone (note: when doing a solution sync, the project cone is all projects).